cmake/KokkosCore_config.h.in

```diff
@@ -55,6 +55,7 @@
 #cmakedefine KOKKOS_OPT_RANGE_AGGRESSIVE_VECTORIZATION  // deprecated
 #cmakedefine KOKKOS_ENABLE_AGGRESSIVE_VECTORIZATION
 #cmakedefine KOKKOS_ENABLE_IMPL_MDSPAN
+#cmakedefine KOKKOS_IMPL_USE_COMPILER_PROVIDED_MDSPAN
 #cmakedefine KOKKOS_ENABLE_IMPL_REF_COUNT_BRANCH_UNLIKELY
```

Comment:
hmm ... should we call it `KOKKOS_USE_COMPILER_PRIVIDED_MDSPAN` or `KOKKOS_ENABLE_IMPL_COMPILER_PROVIDED_MDSPAN`

Reply:
We use `Kokkos_ENABLE*` if it's a `CMake` option. There are no preprocessor variables starting with `KOKKOS_USE`.